_posts/2020-08-08-time-limiting-with-resilience4j.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Implementing Time Limiting with Resilience4j
+categories: [java]
+date: 2020-08-08 05:00:00 +1100
+modified: 2020-08-08 05:00:00 +1100
+author: saajan
+excerpt: "Continuing the Resilience4j journey, this article on TimeLimiter shows when and how to use it to build resilient applications."
+image:
+  auto: 0073-broken
+---
+
+In this series so far, we have learned about Resilience4j and its [Retry](/retry-with-resilience4j/) and [RateLimiter](/rate-limiting-with-resilience4j/) modules. In this article, we will continue exploring Resilience4j with a look into the TimeLimiter. We will find out what problem it solves, when and how to use it, and also look at a few examples.
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/resilience4j/timelimiter" %}
+
+## What is Resilience4j?
+
+Please refer to the description in the previous article for a quick intro into [how Resilience4j works in general](/retry-with-resilience4j/#what-is-resilience4j).
+
+## When to Use the TimeLimiter?
```

Comment:
This section is very focussed on CompletableFuture.

I would like to rename this section to "Time Limiting CompletableFutures" and add another section before it "What are Time Limits for?" (or similar) explaining the concept of timeouts between client and server and why this is necessary. That section doesn't need code, just an explanation of the concept and where we find timelimits in our every day life (there are timeouts on pretty much every level of network communication).

Reply:
@thombergs I have added a new section "What is Time Limiting?", please check.

I am not really sure if we should rename this to "Time Limiting CompletableFutures" though. I renamed it to "When to Use the Resilience4j TimeLimiter?" instead and also added an introductory line to the subsection, please take a look.